backend/infrahub/api/transformation.py

```diff
@@ -92,13 +106,21 @@ async def transform_jinja2(
     transform = await NodeManager.get_one_by_id_or_default_filter(
         db=db,
         id=transform_id,
-        kind=InfrahubKind.TRANSFORMJINJA2,
+        kind=CoreTransformJinja2,
         branch=branch_params.branch,
         at=branch_params.at,
     )
 
-    query = await transform.query.get_peer(db=db)  # type: ignore[attr-defined]
-    repository = await transform.repository.get_peer(db=db)  # type: ignore[attr-defined]
+    query = await transform.query.get_peer(db=db, peer_type=CoreGraphQLQuery, raise_on_error=True)
+    repository = await transform.repository.get_peer(db=db, peer_type=CoreGenericRepository, raise_on_error=True)
+
+    if repository.commit.value is None:  # type: ignore[attr-defined]
+        raise TransformError(
+            repository_name=repository.name.value,
+            location=repository.location.value,
+            commit="n/a",
+            message="Repository doesn't have a commit",
+        )
```

Comment:
I'd move this if-statement to a separate function that gets called by both the Python and Jinja2 transform functions.